pkgs/by-name/ne/nerd-fonts/update.js

```diff
@@ -0,0 +1,72 @@
+const RELEASE_VERSION_FILENAME = "release-version.json";
+const RELEASE_INFO_URL = "https://api.github.com/repos/ryanoasis/nerd-fonts/releases/latest";
+const PKGS_INFO_FILENAME = "fonts.json";
+
+const getFontsInfoUrl = (tag) => `https://raw.githubusercontent.com/ryanoasis/nerd-fonts/refs/tags/${tag}/bin/scripts/lib/fonts.json`;
+
+const fetchJSON = (url) => JSON.parse(std.urlGet(url).split("\r\n").at(-1));
+
+const convertAttrName = (formerAttrName) => {
+    if (/^\d/u.test(formerAttrName)) {
+        return "_" + formerAttrName;
+    } else {
+        return formerAttrName;
+    }
+};
+
+const convertVersion = (formerVersion, dateString) => {
+    if (/^\d/u.test(formerVersion)) {
+        return formerVersion;
+    } else {
+        return "0-unstable-" + dateString;
+    }
+};
+
+let formerReleaseVersion = JSON.parse(std.loadFile(RELEASE_VERSION_FILENAME));
+let releaseInfo = fetchJSON(RELEASE_INFO_URL);
+let tag = releaseInfo.tag_name;
+let releaseVersion = tag.split("v")[1];
+
+if (releaseVersion === formerReleaseVersion) {
+    std.exit(1);
+} else {
+    let fonts = fetchJSON(getFontsInfoUrl(tag)).fonts;
+
+    let pkgsInfo = fonts.map(font => {
+        let pname = font.caskName.toLowerCase();
+
+        let url = releaseInfo.assets
+            .find(asset => asset.name === font.folderName + ".tar.xz")
+            .browser_download_url;
+
+        let sha256 = (() => {
+            let f = std.popen(`nix-prefetch-url ${url} --unpack`, "r");
+            let result = f.getline();
+            f.close();
+            return result;
+        })();
+
+        return ({
+            pname: pname,
+            attrName: convertAttrName(pname),
+            dirName: font.patchedName.replaceAll(/\s+/gu, ""),
+            version: convertVersion(font.version, releaseInfo.published_at.split("T")[0]),
+            longDescription: font.description,
+            license: font.licenseId,
+            url: url,
+            sha256: sha256,
+        });
+    });
+
+    (() => {
+        let f = std.open(PKGS_INFO_FILENAME, "w");
+        f.puts(JSON.stringify(pkgsInfo));
```

Comment:
Maybe this shouldn't matter a lot, but I think it would have been slightly nicer to have The "Reserved Font Name" as an key of each JSON object?